### ISSUE TITLE
WMT Jax does not reach the target

### ISSUE THREAD

**Issue by priyakasimbeg** [mlcommons/algorithmic-efficiency#236]
WMT Jax does not reach the target in the expected number of steps.
## Description
Target Bleu score: 30.879
Target number of steps: 100K

Actual Bleu score: 26.558
Actual Number of steps: 148K

## Steps to Reproduce
on kasimbeg-5  on branch kasimbeg/sacrebleu ran:
```
python3 submission_runner.py \
    --framework=jax \
    --data_dir=/home/kasimbeg/tensorflow_datasets \
    --experiment_dir=/home/kasimbeg \
    --experiment_name=target_setting \
    --workload=wmt \
    --submission_path=reference_algorithms/target_setting_algorithms/jax_adamw.py \
    --tuning_search_space=reference_algorithms/target_setting_algorithms/wmt/tuning_search_space.json > results_wmt_jax.txt 2>&1
```

**Comment by runame** [comment]
This might be because we currently use a different way to calculate the Bleu score (Sourabh started to address this in #211). Did you also compare the validation loss?

**Comment by priyakasimbeg** [comment]
I had patched sourabh's PR in the branch I ran on. I found a number of other diffs in the model architecture and optimizer hparams. Will reconsolidate and rerun.

**Comment by priyakasimbeg** [comment]
The rerun with fixed hparams did also not hit the target (bleu score 30). 
<img width="1349" alt="image" src="https://user-images.githubusercontent.com/12614254/200067860-ddc162de-95ea-439f-b2ab-aa541a4bff7f.png">

**Comment by priyakasimbeg** [comment]
I ran this twice on baejuhan-1 with a newly installed python env from the main branch, and w Juhan's copy of the data and it did not converge. 
Then, I ran it once with juhan's virtual env and it converged. There were a couple of diffs between the packages when I checked but nothing that stood out so have to investigate more.

**Comment by runame** [comment]
@priyakasimbeg Are there any updates on this? The last comment sounds like it does reach the target?

**Comment by priyakasimbeg** [comment]
Sorry, that comment was an update for a different bug.

**Comment by runame** [comment]
> I found a number of other diffs in the model architecture and optimizer hparams. Will reconsolidate and rerun.

Are these already fixed in our main branch? If not, can you tell me the diffs for which you have already tested the fix?
